performance-storage-service/openapi.yaml

```diff
@@ -28,55 +28,132 @@ components:
+        jenkins:
+          $ref: '#/components/schemas/JenkinsMetadata'
+        github:
+          $ref: '#/components/schemas/GithubMetadata'
+        noisepage:
+          $ref: '#/components/schemas/NoisepageMetadata'
+        environment:
+          $ref: '#/components/schemas/EnvironmentMetadata'
+    JenkinsMetadata:
+      description: The metadata related to Jenkins
+      type: object
+      required:
+        - jenkins_job_id
+      properties:
+        jenkins_job_id:
+          type: integer
+          description: The ID of the job which runs this test on Jenkins
+    GithubMetadata:
+      description: The metadata related to GitHub
+      type: object
+      required:
+        - git_commit_id
+        - git_branch
+      properties:
+        git_commit_id:
           type: string
-          description: The name of the test that was run (e.g. TATP)
-        test_settings:
-          $ref: '#/components/schemas/OLTPBenchTestSettings'
-        metrics:
-          $ref: '#/components/schemas/OLTPBenchMetrics'
-    OLTPBenchTestSettings:
+          description: The commit id of the current job
+        git_branch:
+          type: string
+          description: The name of the branch where the nightly job runs
+    NoisepageMetadata:
+      description: The metadata related to Noisepage
+      type: object
+      required:
+        - db_version
+      properties:
+        db_version:
+          type: string
+          description: The version of NoisePage
+    EnvironmentMetadata:
+      description: The metadata related to the environment
+      type: object
+      required:
+        - os_version
+        - cpu_number
+        - cpu_socket
+      properties:
+        os_version:
+          type: string
+          description: The version of the operating system on the testing server.
+        cpu_number:
+          type: string
+          description: The number of CPU used on the testing server
+        cpu_socket:
+          type: string
+          description: The number of the cpu socket on the testing server. When it is larger than 1, this server is a NUMA system
+    OLTPBenchParameters:
       description: The settings that the test was run with
       type: object
       required:
+        - query_mode
         - scale_factor
         - terminals
+        - client_time
         - transaction_weights
       properties:
+        query_mode:
+          type: string
+          enum:
+            - simple
+            - extended
+          description: The query mode when running the testing script.
         scale_factor:
           type: number
           format: float
-          description: A scale factor for the size of the database
+          description: The size of the database to load.
         terminals:
           type: integer
+          description: The number of client threads that will issue requests to the DBMS.
+        client_time:
+          type: integer
+          enum:
+            - 60
+            - 600
```

Comment:
I don't think we are enforcing this right? it can be any integer

Reply:
Done.